src/game/component/_private/PhysicsComponent.cpp

```diff
@@ -37,6 +45,229 @@ bool PhysicsComponent::Impulse()
+	m_HitboxEntity.Add<MovableComponent>();
+	m_HitboxEntity.Add<MaterialComponent>()->SetOpacity(0.5f);
+	DrawComponent *pDraw = m_HitboxEntity.Add<DrawComponent>();
+	pDraw->SetGeometry(ShapeType::PLANE);
+	pDraw->SetTexture(TEXTURE_PATH + "green.tga", TextureType::RGB);
+
+	PhysicsSystem::GetWorld()->addRigidBody(m_pBody->body);
+
+	toggleHitboxView();
+}
+
+
+
+void PhysicsComponent::addSphere(float mass, float additionToRadius)
+{
+	float f[3] = { m_pTransform->GetScale()[0], m_pTransform->GetScale()[1], m_pTransform->GetScale()[2] };
+	float radius = std::max(std::max(f[1], f[2]), f[3]) + additionToRadius;
+	float X = m_pTransform->GetPosition()[0];
+	float Y = m_pTransform->GetPosition()[1];
+	float Z = m_pTransform->GetPosition()[2];
+
+	btTransform transform;
+	transform.setIdentity();
+
+	transform.setOrigin(btVector3(X, Y, Z));
+
+	btSphereShape *sphere = new btSphereShape(radius);
+
+	btVector3 inertia(0, 0, 0);
+	if (mass != 0.0)
+	{
+		sphere->calculateLocalInertia(mass, inertia);
+	}
+
+	btMotionState *motion = new btDefaultMotionState(transform);
+
+	btRigidBody::btRigidBodyConstructionInfo info(mass, motion, sphere, inertia);//note that we passed mass to not make it static
+	btRigidBody *body = new btRigidBody(info);
+
+	body->setRestitution(0);
+
+	m_pBody = new bulletObject(body, btVector3(1, 1, 1), btVector3(1, 1, 1));
+
+	m_HitboxEntity = EntityManager::CreateEntity();
+	m_HitboxEntity.Add<TransformComponent>()->Init(glm::vec3(X, Y, Z), glm::vec3(radius, radius, radius));
+	m_HitboxEntity.Add<MovableComponent>();
+	m_HitboxEntity.Add<MaterialComponent>()->SetOpacity(0.5f);
+	DrawComponent *pDraw = m_HitboxEntity.Add<DrawComponent>();
+	pDraw->SetGeometry(ShapeType::PLANE);
+	pDraw->SetTexture(TEXTURE_PATH + "circle.tga", TextureType::RGBA);
+
+	PhysicsSystem::GetWorld()->addRigidBody(m_pBody->body);
+
+	toggleHitboxView();
+}
+
+
+
+void PhysicsComponent::addCylinder(float mass, float additionTodiameter, float additionToHeight)
+{
+	float X = m_pTransform->GetPosition()[0];
+	float Y = m_pTransform->GetPosition()[1];
+	float Z = m_pTransform->GetPosition()[2];
+	float diameter = m_pTransform->GetScale()[0] + additionTodiameter;
+	float height = m_pTransform->GetScale()[1] + additionToHeight;
+
+	btTransform transform;
+	transform.setIdentity();
+	transform.setOrigin(btVector3(X, Y, Z));
+
+	btCylinderShape *cylinder = new btCylinderShape(btVector3(diameter, height, diameter));
+
+	btVector3 inertia(0, 0, 0);
+	if (mass != 0.0)
+	{
+		cylinder->calculateLocalInertia(mass, inertia);
+	}
+
+	btMotionState *motion = new btDefaultMotionState(transform);
+
+	btRigidBody::btRigidBodyConstructionInfo info(mass, motion, cylinder, inertia);
+	btRigidBody *body = new btRigidBody(info);
+
+	body->setRestitution(0);
+
+	m_pBody = new bulletObject(body, btVector3(1, 1, 1), btVector3(1, 1, 1));
+
+	PhysicsSystem::GetWorld()->addRigidBody(m_pBody->body);
```

Comment:
I actually started with the queue method. The problem with that was wasting time on every tick and checking whether the current body has already been added to the world or not.